### DISCUSSION TITLE
Ability to turn off tab synchronization

### DISCUSSION THREAD

**Discussion by shishkin17** [pmndrs/jotai#2580] [Ideas]
Hello,

It turns out that if I don't what tab synchronization, I need to create the whole storage object (without subscription) even though I am ok with the built in implementation.
It would be great to have an option just not to subscribe.

Thanks!

**Comment by dai-shi** [comment]
With v2.8.2, we can customize `subscribe`, so a workaround is to provide a noop subscribe.
#2539

```js
import { createJSONStorage, atomWithStorage } from 'jotai/utils';

const myStorage = createJSONStorage(() => ({ ...localStorage, subscribe: () => () => {} }));
const myAtom = atomWithStorage('foo', 'foo', myStorage);
```

**Reply by shishkin17** [reply]
Thank you for suggestion, but unfortunately this is not working. Destructuring localStorage doesn't copy `getItem()` and it fails.
I tried to make a copy of localStorage with `Object.assign()`, it didn't fail, but `subscribe` was added as a field to storage so it didn't work.

Solution that finally worked for me is:

```
const myStorage = createJSONStorage();
myStorage.subscribe = undefined;
const myAtom = atomWithStorage('foo', 'foo', myStorage);
```

I suggest adding this to docs on how not to subscribe, because it is not obvious at the beginning.

**Reply by dai-shi** [reply]
`myStorage.subscribe = undefined;` oh, that's much easier!!!
Yes, please open a PR to improve the docs.

**Reply by shishkin17** [reply]
I ran into weird problems because of that, so I suggest not to remove subscription.
